openmls/src/group/tests/external_remove_proposal.rs

```diff
@@ -0,0 +1,449 @@
+use openmls_rust_crypto::OpenMlsRustCrypto;
+use rstest::*;
+use rstest_reuse::{self, *};
+
+use crate::{
+    framing::*,
+    group::{config::CryptoConfig, errors::*, *},
+    messages::external_proposals::*,
+    prelude::ProtocolVersion,
+};
+
+use openmls_traits::types::Ciphersuite;
+
+use super::utils::*;
+
+// Creates a standalone group
+fn new_test_group(
+    identity: &str,
+    wire_format_policy: WireFormatPolicy,
+    ciphersuite: Ciphersuite,
+    backend: &impl OpenMlsCryptoProvider,
+) -> (MlsGroup, CredentialWithKeyAndSigner) {
+    let group_id = GroupId::from_slice(b"Test Group");
+
+    // Generate credential bundles
+    let credential_with_keys =
+        generate_credential_bundle(identity.into(), ciphersuite.signature_algorithm(), backend);
+
+    // Define the MlsGroup configuration
+    let mls_group_config = MlsGroupConfig::builder()
+        .wire_format_policy(wire_format_policy)
+        .crypto_config(CryptoConfig::with_default_version(ciphersuite))
+        .build();
+
+    (
+        MlsGroup::new_with_group_id(
+            backend,
+            &credential_with_keys.signer,
+            &mls_group_config,
+            group_id,
+            credential_with_keys.credential_with_key.clone(),
+        )
+        .unwrap(),
+        credential_with_keys,
+    )
+}
+
+// Validation test setup
+fn validation_test_setup(
```

Comment:
You (almost) never use this helper although you always need in your tests a group with Alice & Bob